examples/platform/infineon/trustm/DeviceAttestationCredsExampleTrustM.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "DeviceAttestationCredsExampleTrustM.h"
+
+#include <credentials/examples/ExampleDACs.h>
+#include <credentials/examples/ExamplePAI.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+
+#if CHIP_CRYPTO_HSM
+#include <crypto/hsm/CHIPCryptoPALHsm.h>
+#endif
+
+#ifdef ENABLE_HSM_DEVICE_ATTESTATION
+
+#include <crypto/hsm/infineon/CHIPCryptoPALHsm_utils_trustm.h>
+
+/* Device attestation key ids for Trust M */
+#define DEV_ATTESTATION_KEY_ID 0xE0F0
+#define DEV_ATTESTATION_CERT_ID 0xE0E3
+#define PAI_CERT_ID 0xE0E8
+#define CERT_DECLARATION_ID 0xF1E0
+
+namespace chip {
+namespace Credentials {
+namespace Examples {
+
+namespace {
+
+class ExampleTrustMDACProvider : public DeviceAttestationCredentialsProvider
+{
+public:
+    CHIP_ERROR GetCertificationDeclaration(MutableByteSpan & out_cd_buffer) override;
+    CHIP_ERROR GetFirmwareInformation(MutableByteSpan & out_firmware_info_buffer) override;
+    CHIP_ERROR GetDeviceAttestationCert(MutableByteSpan & out_dac_buffer) override;
+    CHIP_ERROR GetProductAttestationIntermediateCert(MutableByteSpan & out_pai_buffer) override;
+    CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & message_to_sign, MutableByteSpan & out_signature_buffer) override;
+};
+
+CHIP_ERROR ExampleTrustMDACProvider::GetDeviceAttestationCert(MutableByteSpan & out_dac_buffer)
+{
+#if 0
```

Comment:
Why are all these `if 0` here?

Reply:
Hi @bzbarsky-apple, Thanks for your review. We have removed it in the latest commit.